File: leetcode/java/27.RemoveElement.java
class Solution {
  public int removeElement(int[] nums, int val) {
      int start = 0, end = nums.length -1;
      while (start <= end) {
          if (nums[start] != val) {
              start ++;
              continue;
          }
          if (nums[end] == val) {
              end --;
              continue;
          }
          nums[start] = nums[end];
          start ++;
          end --;
      }
      return start;
  }
}

// import java.util.*;
// import java.util.stream.Collectors;
// class Main
//   public static int removeElement(int[] nums,int val){
//    List<Integer> arr = Arrays.stream(nums).boxed().collect(Collectors.toList());
//    arr = arr.stream().filter(x->x!=val)
//    .collect(Collectors.toList())
//    ;
//    // Integer[] x = arr.toArray();
//    nums = arr.stream().mapToInt(i->i)   .toArray();
//   System.out.println(nums[0]+" "+nums[1]);
//    return arr.size();
//   }

//   public static void main(String ...s){
//     System.out.println(removeElement(new int[]{3,2,2,3},3));
//   }

// }

File: leetcode/kotlin/27.RemoveElement.kt
import java.util.*
import java.util.stream.Collectors

internal object Main {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var nums = nums
        var arr = Arrays.stream(nums).boxed().collect(Collectors.toList())
        arr = arr.stream().filter { x: Int -> x != `val` }
            .collect(Collectors.toList())
        // Integer[] x = arr.toArray();
        nums = arr.stream().mapToInt { i: Int? -> i!! }.toArray()
        println(nums[0].toString() + " " + nums[1])
        return arr.size
    }

    @JvmStatic
    fun main(s: Array<String>) {
        println(removeElement(intArrayOf(3, 2, 2, 3), 3))
    }
}